### PR TITLE
Feature/trlst 171 func tests

### PR DESCRIPTION
**Pull request by davecharles**
This PR partially implements [TRLST-171](https://uktrade.atlassian.net/browse/TRLST-171?atlOrigin=eyJpIjoiZjA0YWE2NjNiYjQ0NDk3Njk3ZjYwNWY2ODg1OGRmOGEiLCJwIjoiaiJ9)

Additional changes are required on [trade-remedies-docker](https://github.com/uktrade/trade-remedies-docker).

This change introduces [behave-django](https://behave-django.readthedocs.io/en/stable/index.html) tests to trade remedies public portal. Main changes to note:

 - Dockerize added to Dockerfile definition.
 - `behave-django` and selenium added to deps.
 - Updated `local.py` settings to include behave additions. 
 - Added new `features` folder at `manage.py` level  which includes:
   - `environment.py`: behave requirement for config of setup/teardown steps.
   - `.feature` files.
   -  supporting `steps` python modules.
 - Simple BDD tests added for portal landing pages.

### CODE REVIEW

File: requirements.in/dev.in [hunk 1]
```diff
@@ -1,11 +1,13 @@
 -r base.in
 
-ipython==7.10
-traitlets==4.3.3
+behave-django==1.4.0
 black
-flake8
 coverage
+flake8
+ipython==7.10
```
Comment (kevinetienne) [inline requirements.in/dev.in:7]: looks like latest version is `7.19` any reason to stick to `7.10`

File: trade_remedies_public/features/environment.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+
+def before_feature(context, feature):  # no-qa
+    configure(context)
+
+
+def after_step(context, step):
+    if step.status == 'failed':
+        context.browser.save_screenshot(
+            f'/code/test-reports/bdd-screenshots/{context.scenario.name}-{step.name}.png'
```
Comment (kevinetienne) [inline trade_remedies_public/features/environment.py:39]: When there's a failure inside docker, do you have to copy the `png` or is the folder shared with the host?

File: requirements.in/dev.in [hunk 3]
```diff
@@ -1,11 +1,13 @@
 -r base.in
 
-ipython==7.10
-traitlets==4.3.3
+behave-django==1.4.0
```
Comment (kevinetienne) [inline requirements.in/dev.in:3]: Would it be nice to hook to tests with circle ci. Maybe you are planning to do it after this is PR is merged?

**Review by kevinetienne — APPROVED** [review]
Looks good to me but I haven't worked on trade remedies, maybe someone else can check it too as I might be ignorant of the infrastructure.

File: trade_remedies_public/features/environment.py (hunk 2 above)
Comment (davecharles) [reply trade_remedies_public/features/environment.py:39]: Good question, it should write to shared host folder `test-reports` (the xml run log is). Will try to force fail/screenshot creation to confirm.

File: requirements.in/dev.in (hunk 3 above)
Comment (davecharles) [reply requirements.in/dev.in:3]: Yep - was planning to get this running locally first then roll into CI in the future.

File: requirements.in/dev.in (hunk 1 above)
Comment (davecharles) [reply requirements.in/dev.in:7]: I basically mimicked what ICMS/Fido do - I'll bump and give it a try :)

File: Dockerfile [hunk 4]
```diff
@@ -1,7 +1,14 @@
...
 WORKDIR /app
 
 RUN pip install --upgrade pip
-RUN pip install -r requirements/dev.txt
\ No newline at end of file
+RUN pip install -r requirements/dev.txt
+
+# Install dockerize
+RUN \
+  wget https://github.com/jwilder/dockerize/releases/download/$DOCKERIZE_VERSION/dockerize-alpine-linux-amd64-$DOCKERIZE_VERSION.tar.gz \
```
Comment (rossmiller) [inline Dockerfile:12]: @davecharles what is dockerize used for here?

File: requirements.in/dev.in (hunk 1 above)
Comment (rossmiller) [reply requirements.in/dev.in:7]: Not sure if this is because of traitlets needing to be pegged to 4.3.3

File: requirements.in/dev.in [hunk 5]
```diff
@@ -1,11 +1,13 @@
...
+flake8
+ipython==7.10
 pytest
 pytest-cov
 pytest-django
-pytest-xdist
\ No newline at end of file
+pytest-xdist
+selenium==3.141.0
+traitlets==4.3.3
```
Comment (rossmiller) [inline requirements.in/dev.in:13]: Maybe we could bin traitlets - I am certainly not using it.

**Review by rossmiller — COMMENTED** [review]
A couple of comments.

File: requirements.in/dev.in (hunk 1 above)
Comment (davecharles) [reply requirements.in/dev.in:7]: Latest is `IPython 7.12: Released on Jan 31st 2020`, have bumped and seems to work with existing deps.

File: trade_remedies_public/features/environment.py (hunk 2 above)
Comment (davecharles) [reply trade_remedies_public/features/environment.py:39]: Good spot @kevinetienne, the path was (a) wrong and (b) needs to exist, will commit a fix.

Comment (davecharles) [reply trade_remedies_public/features/environment.py]: Fixed with 64e080d

File: requirements.in/dev.in (hunk 5 above)
Comment (davecharles) [reply requirements.in/dev.in:13]: Maybe we can look at that as part of the re-org?

**Review by rossmiller — APPROVED** [review]
LGTM!

**Pull request merged**